apps/web/src/services/asset-inventory/collector/modules/CollectorContents.vue

```diff
@@ -0,0 +1,194 @@
+
+import CollectorItemInfo from '@/services/asset-inventory/collector/modules/CollectorItemInfo.vue';
+import CollectorListNoData from '@/services/asset-inventory/collector/modules/CollectorListNoData.vue';
+import { CollectorItemInfoType } from '@/services/asset-inventory/collector/type';
+import ProviderList from '@/services/asset-inventory/components/ProviderList.vue';
+import { ASSET_INVENTORY_ROUTE } from '@/services/asset-inventory/route-config';
+import { useCollectorPageStore } from '@/services/asset-inventory/store/collector-page-store';
+
+interface Props {
+    queryHelper?: ApiQueryHelper
+    totalCount?: number
+    pageLimit?: number
+}
+
+const props = withDefaults(defineProps<Props>(), {
+    queryHelper: undefined,
+    totalCount: 0,
+    pageLimit: 15,
+});
+
+const cloudCollectorPageStore = useCollectorPageStore();
+const cloudCollectorPageState = cloudCollectorPageStore.$state;
+
+const state = reactive({
+    providerList: [] as ReferenceItem[],
+    searchTags: [] as QueryTag[],
+    selectedProvider: computed(() => cloudCollectorPageState.selectedProvider),
+});
+
+const handlerState = reactive({
+    infoItems: [
+        { key: CollectorItemInfoType.PLUGIN, label: 'Plugin' },
+        { key: CollectorItemInfoType.STATUS, label: 'Current Status' },
+        { key: CollectorItemInfoType.JOBS, label: 'Recent Collector Jobs' },
+        { key: CollectorItemInfoType.SCHEDULE, label: 'Schedule' },
+    ],
+    excelFields: [
+        { key: 'name', name: 'Name' },
+        { key: 'state', name: 'State' },
+        { key: 'plugin_info.plugin_id', name: 'Plugin' },
+        { key: 'plugin_info.version', name: 'Version' },
+        { key: 'last_collected_at', name: 'Last Collected', type: 'datetime' },
+    ],
+});
+
+/* Components */
+const handleSelectedProvider = (providerName: string) => {
+    cloudCollectorPageStore.setSelectedProvider(providerName);
+};
+const handleCreate = () => {
+    SpaceRouter.router.push({ name: ASSET_INVENTORY_ROUTE.COLLECTOR.CREATE._NAME });
+};
+const handleExport = async () => {
+    await store.dispatch('file/downloadExcel', {
+        url: '/inventory/collector/list',
+        param: { query: props.queryHelper.data },
+        fields: handlerState.excelFields,
+        file_name_prefix: FILE_NAME_PREFIX.collector,
+    });
+};
+const handleChange = async () => {
+    // setApiQueryWithToolboxOptions(collectorApiQueryHelper, options);
+    // if (options.queryTags !== undefined) {
+    //     state.searchTags = options.queryTags;
+    //     queryHelper.setFiltersAsQueryTag(options.queryTags);
+    //     await replaceUrlQuery('filters', queryHelper.rawQueryStrings);
+    // }
+    // await listCollectors();
+};
+
+/* Watcher */
+watch(() => store.state.reference.provider.items, (value: ReferenceItem) => {
+    state.providerList = [
+        { key: 'all', name: 'All Providers' },
+        ...Object.values(value),
+    ];
+}, { immediate: true });
```

Comment:
I think it would be better to get computed value rather than watching the store.

```suggestion
const storeState = reactive({
  providers: computed<ProviderReferenceMap>(() => store.getters['reference/providerItems']),
});
const state = reactive({
  providerList: computed(() => ([{ key: 'all', name: 'All Providers' }, ...Object.values(storeState.providers) ])),
});
```